mithril-test-lab/mithril-end-to-end/src/assertions/check.rs

```diff
@@ -161,6 +162,84 @@ pub async fn assert_signer_is_signing_snapshot(
     }
 }
 
+pub async fn assert_node_producing_cardano_database_snapshot(
+    aggregator_endpoint: &str,
+) -> StdResult<String> {
+    let url = format!("{aggregator_endpoint}/artifact/cardano-database");
+    info!("Waiting for the aggregator to produce a Cardano database snapshot");
+
+    // todo: reduce the number of attempts if we can reduce the delay between two immutables
+    match attempt!(45, Duration::from_millis(2000), {
+        match reqwest::get(url.clone()).await {
```

Comment:
I found that's a bit complicated to identify the content of the attempt! with the 2 matchs in 2 lines. 
It could be simpler to read if we create a function called by the attemp! (if it's possible).

Reply:
I have created a sub-function for the content of the `attempt!`. I also modified all the other assertions for consistency :+1: